python_modules/libraries/dagster-airbyte/dagster_airbyte_tests/experimental/test_resources.py

```diff
@@ -144,3 +158,226 @@ def test_basic_resource_request(
+                status=AirbyteJobStatusType.SUCCEEDED
+                if succeed_at_end
+                else AirbyteJobStatusType.FAILED
+            ),
+            status=200,
+        )
+        return client.sync_and_poll(connection_id=TEST_CONNECTION_ID, poll_interval=0.1)
+
+    if succeed_at_end:
+        assert _mock_interaction() == AirbyteOutput(
+            job_details=get_job_details_sample(AirbyteJobStatusType.SUCCEEDED),
+            connection_details=SAMPLE_CONNECTION_DETAILS,
+        )
+    else:
+        with pytest.raises(Failure, match="Job failed"):
+            _mock_interaction()
+
+
+@pytest.mark.parametrize(
+    "cancel_on_termination",
+    [
+        True,
+        False,
+    ],
+    ids=[
+        "cancel_on_termination_true",
+        "cancel_on_termination_false",
+    ],
+)
+def test_airbyte_sync_and_poll_client_cancel_on_termination(
+    cancel_on_termination: bool, base_api_mocks: responses.RequestsMock
+) -> None:
+    resource = AirbyteCloudWorkspace(
+        workspace_id=TEST_WORKSPACE_ID,
+        client_id=TEST_CLIENT_ID,
+        client_secret=TEST_CLIENT_SECRET,
+    )
+    client = resource.get_client()
+
+    test_job_endpoint = f"jobs/{TEST_JOB_ID}"
+    test_job_api_url = f"{AIRBYTE_REST_API_BASE}/{AIRBYTE_REST_API_VERSION}/{test_job_endpoint}"
+
+    # Create mock responses to mock full sync and poll behavior to test statuses, used only in this test
+    base_api_mocks.add(
+        method=responses.POST,
+        url=f"{AIRBYTE_REST_API_BASE}/{AIRBYTE_REST_API_VERSION}/jobs",
+        json=get_job_details_sample(status=AirbyteJobStatusType.PENDING),
+        status=200,
+    )
+    base_api_mocks.add(
+        method=responses.GET,
+        url=test_job_api_url,
+        json=get_job_details_sample(status=TEST_UNRECOGNIZED_AIRBYTE_JOB_STATUS_TYPE),
+        status=200,
+    )
+    base_api_mocks.add(
+        method=responses.POST,
+        url=f"{AIRBYTE_CONFIGURATION_API_BASE}/{AIRBYTE_CONFIGURATION_API_VERSION}/connections/get",
+        json=SAMPLE_CONNECTION_DETAILS,
+        status=200,
+    )
+
+    if cancel_on_termination:
+        base_api_mocks.add(
+            method=responses.DELETE,
+            url=test_job_api_url,
+            status=200,
+            json=get_job_details_sample(status=AirbyteJobStatusType.CANCELLED),
+        )
+
+    with pytest.raises(Failure, match="unexpected state"):
+        client.sync_and_poll(
+            connection_id=TEST_CONNECTION_ID,
+            poll_interval=0,
+            cancel_on_termination=cancel_on_termination,
+        )
+    if cancel_on_termination:
+        assert_rest_api_call(
+            call=base_api_mocks.calls[-1], endpoint=test_job_endpoint, method=responses.DELETE
+        )
+    else:
+        # If we don't cancel on termination, the last call will be a call to fetch the job details
+        assert_rest_api_call(
+            call=base_api_mocks.calls[-1], endpoint=test_job_endpoint, method=responses.GET
+        )
```

Comment:
really don't love these complex ternaries in test body IMO. Just really hard to read, I think this should just be a part of the parameterization.

Reply:
That's a good call - updated in [5df711b](https://github.com/dagster-io/dagster/pull/26431/commits/5df711bd399de0494fed1d6a6ad2e0296072ac1c)